models/nft/ethereum/nft_ethereum_top_erc721_holders.sql

```diff
@@ -0,0 +1,47 @@
+{{ config(
+       alias = 'top_erc721_holders',
+       materialized='table',
+       post_hook='{{ expose_spells(\'["ethereum"]\',
+                                   "sector",
+                                   "nft",
+                                   \'["Henrystats"]\') }}'
+       )
+   }}
+
+WITH 
+
+erc721_balances as (
+    SELECT 
+        wallet_address,
+        token_address as nft_contract_address,
+        COUNT(tokenId) as balance 
+    FROM 
+    {{ ref('balances_ethereum_erc721_latest') }}
+    GROUP BY 1, 2
+), 
+
+total_supply as (
+    SELECT 
+        wallet_address, 
+        nft_contract_address, 
+        balance, 
+        SUM(balance) OVER (PARTITION BY nft_contract_address) as total_supply
+    FROM 
+    erc721_balances
+)
+
+SELECT 
+    * 
+FROM 
+(
+ SELECT 
+     wallet_address,
+     nft_contract_address, 
+     balance, 
+     balance/total_supply as supply_share,
+     total_supply, 
+    ROW_NUMBER() OVER (PARTITION BY nft_contract_address ORDER BY balance DESC) as rn
+ FROM 
+ total_supply
+) x 
+-- WHERE rn <= 50 
```

Comment:
do you want to keep this to more than 50? then filter on the app instead?

Reply:
hmm there's already a balances table soo I guess it's better to keep it as top 50. But what would you suggest? happy with either

Reply:
updated